SpatialGDK/Source/Private/EngineClasses/SpatialGameInstance.cpp

```diff
@@ -51,45 +52,52 @@ bool USpatialGameInstance::HasSpatialNetDriver() const
 	return bHasSpatialNetDriver;
 }
 
-bool USpatialGameInstance::StartGameInstance_SpatialGDKClient(FString& Error)
+bool USpatialGameInstance::StartGameInstance_SpatialGDKClient(bool bIsPIE, FString& Error)
 {
-	if (WorldContext->PendingNetGame)
+	FURL URL;
+	if (bIsPIE)
 	{
-		if (WorldContext->PendingNetGame->NetDriver && WorldContext->PendingNetGame->NetDriver->ServerConnection)
-		{
-			WorldContext->PendingNetGame->NetDriver->ServerConnection->Close();
-			GetEngine()->DestroyNamedNetDriver(WorldContext->PendingNetGame, WorldContext->PendingNetGame->NetDriver->NetDriverName);
-			WorldContext->PendingNetGame->NetDriver = nullptr;
-		}
-
-		WorldContext->PendingNetGame = nullptr;
+		// Do this so we load the same world that the editor had running.
+		URL = WorldContext->LastURL;
 	}
+	else
+	{
+		URL = GetInitialGameURL();
+	}
+	// If the default is just a map, set the URL to localhost to make sure we attempt to make a connection.
+	URL.Host = SpatialConstants::LOCAL_HOST;
 
-	// Clean up the netdriver/socket so that the pending level succeeds
-	if (GetWorldContext()->World())
+	EBrowseReturnVal::Type BrowseResult = EBrowseReturnVal::Failure;
+
+	if (URL.Valid)
 	{
-		GetEngine()->ShutdownWorldNetDriver(GetWorldContext()->World());
+		BrowseResult = GEngine->Browse(*WorldContext, URL, Error);
 	}
 
-	FURL URL = WorldContext->LastURL;
-	URL.Host = SpatialConstants::LOCAL_HOST;
+	return BrowseResult != EBrowseReturnVal::Failure;
+}
 
-	WorldContext->PendingNetGame = NewObject<USpatialPendingNetGame>();
-	WorldContext->PendingNetGame->Initialize(URL);
-	WorldContext->PendingNetGame->InitNetDriver();
-	bool bOk = true;
+FURL USpatialGameInstance::GetInitialGameURL() const
+{
+	// Load the default URL from config files.
+	FURL DefaultURL;
+	DefaultURL.LoadURLConfig(TEXT("DefaultPlayer"), GGameIni);
+
+	const TCHAR* Cmd = FCommandLine::Get();
+
+	// Load up the default game map settings.
+	const UGameMapsSettings* GameMapsSettings = GetDefault<UGameMapsSettings>();
```

Comment:
Can't this be accessed in the if statement as both `GameMapsSettings` and `DefaultMap` are only used in within that scope?